src/main/java/seedu/smarthomebot/logic/commands/ListCommand.java

```diff
@@ -67,32 +83,35 @@ private CommandResult listAppliance() {
 
             if (filterApplianceList.isEmpty()) {
                 if (locationList.isLocationCreated(filteredLocation)) {
-                    return new CommandResult("There is no appliance in \"" + filteredLocation + "\".");
+                    throw new NoApplianceInLocationException();
                 }
-                return new CommandResult("Location: \"" + filteredLocation + "\" does not exist.");
+                throw new LocationNotFoundException();
             }
             String header = (LINE + "Here are the appliances in \"" + filteredLocation + "\"");
-            String outputFilteredList = displayOutput(header, filterApplianceList);
-            return new CommandResult(outputFilteredList);
+            outputResults = displayOutput(header, filterApplianceList);
         }
+        return new CommandResult(outputResults);
     }
 
-    private CommandResult listLocation(int index) {
-        if (locationList.getAllLocations().size() == 0) {
-            return new CommandResult(LINE + MESSAGE_LIST_NO_LOCATIONS);
+    private CommandResult listLocation(int index) throws EmptyLocationListException {
+
+        ArrayList<String> listLocationList = locationList.getAllLocations();
```

Comment:
Should we rename this to something clearer? Like toPrintLocationList or outputLocationList?